addons/l10n_in_edi_ewaybill/models/account_edi_format.py

```diff
@@ -105,25 +105,20 @@ def _check_move_configuration(self, move):
         error_message += self._l10n_in_validate_partner(move.partner_id)
         error_message += self._l10n_in_validate_partner(move.company_id.partner_id, is_company=True)
         if not re.match("^.{1,16}$", is_purchase and move.ref or move.name):
-            error_message.append(_("%s number should be set and not more than 16 characters",
+            error_message.append(_("%s number can not be more than 16 characters",
                 (is_purchase and "Bill Reference" or "Invoice")))
         goods_line_is_available = False
-        for line in move.invoice_line_ids.filtered(lambda line: not (line.display_type in ('line_section', 'line_note', 'rounding') or line.product_id.type == "service")):
+        empty_hsn_indices = []
+        goods_line_ids = move.invoice_line_ids.filtered(lambda line: line.display_type not in {'line_section', 'line_note', 'rounding'}).sorted(lambda l: (l.sequence, l.id))
+        for line_number, line in enumerate(goods_line_ids, start=1):
```

Comment:
```suggestion
        empty_hsn_indices = []
        goods_lines = move.invoice_line_ids.filtered(lambda line: line.display_type not in {'line_section', 'line_note', 'rounding'}).sorted(lambda l: (l.sequence, l.id))
        for line_number, line in enumerate(goods_line_ids, start=1):
```
if you end a variable with _id(s) it means it just contain an integer or a sequence of integer, not a recordset